### PR TITLE
docs: Fix git-format-staged in precommit documentation

### PR DESCRIPTION
**Pull request by Pietrucci-Blacher**
Hello, here is a little proposal to clean some deprecated stuff in the git-format-staged section on the pre-commit page.

## Description
I deleted / replaced old commands that have been deprecated or removed in Husky.

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: docs/precommit.md [hunk 1]
```diff
@@ -72,25 +72,25 @@ Git-format-staged requires Python v3 or v2.7. Python is usually pre-installed on
 <!--npm-->
 
 ```bash
-npx husky-init
+npx husky init
 npm install --save-dev git-format-staged
-npx husky set .husky/pre-commit "git-format-staged -f 'prettier --ignore-unknown --stdin --stdin-filepath \"{}\"' ."
+echo "git-format-staged -f 'prettier --ignore-unknown --stdin --stdin-filepath \"{}\"' ." > .husky/pre-commit
```
Comment (fisker) [inline docs/precommit.md:77]: Please use `fs.writeFileSync` to write file.

https://github.com/prettier/prettier/pull/16000#discussion_r1477057609

**Review by fisker — APPROVED** [review]

File: docs/precommit.md [hunk 2]
```diff
@@ -57,33 +57,33 @@ Git-format-staged requires Python v3 or v2.7. Python is usually pre-installed on
 <!--npm-->
 
 ```bash
-npx husky-init
+npx husky init
 npm install --save-dev git-format-staged
-npx husky set .husky/pre-commit "git-format-staged -f 'prettier --ignore-unknown --stdin --stdin-filepath \"{}\"' ."
+node --eval "fs.writeFileSync('.husky/pre-commit', 'git-format-staged -f \\'prettier --ignore-unknown --stdin --stdin-filepath \"{}\"\\' .\\n')"
```
Comment (fisker) [inline docs/precommit.md:62]: This command seems not excuteable. 

```
>node --eval "fs.writeFileSync('.husky/pre-commit', 'git-format-staged -f \\'prettier --ignore-unknown --stdin --stdin-filepath \"{}\"\\' .\\n')"
[eval]:1
fs.writeFileSync('.husky/pre-commit', 'git-format-staged -f \\'prettier --ignore-unknown --stdin --stdin-filepath "{}"\\' .\\n')
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^

SyntaxError: missing ) after argument list
```

replace `\\` with `\` seems works

Comment (Pietrucci-Blacher) [reply docs/precommit.md:62]: Hi, yes, it works, but isn't there a risk of compatibility problems with Windows if you just put a backslash?

Comment (fisker) [reply docs/precommit.md:62]: The error above was thrown when I ran on Windows (cmd.exe).

**Review by fisker — APPROVED** [review]

**Pull request merged**